### PR TITLE
feat: added gif support

### PR DESCRIPTION
**Pull request by benswinburne**

### CODE REVIEW

File: src/handler.js [hunk 1]
```diff
@@ -65,7 +67,7 @@ const handle = async message => {
 
   const { filetype } = metadata.file;
 
-  if (!['png', 'jpeg', 'jpg'].includes(filetype)) {
+  if (!['png', 'jpeg', 'jpg', 'gif'].includes(filetype)) {
```
Comment (guyfedwards) [inline src/handler.js]: Need to move the `isEmojiMsg` check logic above this

File: src/handler.js [hunk 2]
```diff
@@ -99,12 +102,15 @@ const handle = async message => {
...
-      withoutEnlargement: true,
-    });
+  const resizer =
+    metadata.file.mimetype === 'image/gif'
+      ? new Gifsicle(['--resize-fit', '128'])
+      : sharp()
+          .max()
+          .resize(128, 128, {
+            fit: sharp.fit.inside,
+            withoutEnlargement: true,
```
Comment (guyfedwards) [inline src/handler.js]: ┴┬┴┤( ͡° ͜ʖ├┬┴┬

**Review by guyfedwards — APPROVED** [review]

**Review by guyfedwards — CHANGES_REQUESTED** [review]
Move `isEmojiMsg` above filetype check

**Comment by guyfedwards** [comment]
Can you also test this with https://github.com/guyfedwards/emoji/blob/master/emoji/asdf.gif as thats the gif i was having issues with previously

**Comment by benswinburne** [comment]
Will do

**Comment by benswinburne** [comment]
@guyfedwards 	https://github.com/guyfedwards/emoji/blob/test-upload/emoji/asdfyoyoodad.gif looks good

**Review by guyfedwards — APPROVED** [review]
:+1: looks good. Would be nice next step to move handler and verification into own files I think

**Comment by benswinburne** [comment]
Yeah i started that in my tests branch but didn't go further as i was waiting for your github branch. I'll pr the test one in a sec

**Pull request merged**